File: src/MixNetFx/ADSqlSupportComponents/AdSqlDirectoryPickerDialog.vb
Public Class ADSqlDirectoryPickerDialog

    Private mySqlPathFilename As String

    Public Overloads Function ShowDialog(ByVal ServerConnection As String, ByVal ExtensionFilter As String) As DialogResult

        With DirectoryPicker
            .BeginUpdate()
            .ConnectionString = ServerConnection
            .ExtensionFilter = ExtensionFilter
            .EndUpdate()
        End With
        Return Me.ShowDialog()
    End Function

    Public Overloads Function ShowDialog(ByVal ServerConnection As String) As DialogResult

        With DirectoryPicker
            .BeginUpdate()
            .ConnectionString = ServerConnection
            .ExtensionFilter = Nothing
            .EndUpdate()
        End With
        Return Me.ShowDialog()
    End Function

    Private Sub DirectoryPicker_SelectedFileNodeChanged(ByVal sender As Object, ByVal e As ADFileTreeViewEventArgs) Handles DirectoryPicker.SelectedFileNodeChanged
        txtPath.Text = e.Node.FullPath
    End Sub

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        mySqlPathFilename = txtPath.Text
    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        mySqlPathFilename = Nothing
    End Sub

    Public Property SqlPathFilename() As String
        Get
            Return mySqlPathFilename
        End Get
        Set(ByVal value As String)
            mySqlPathFilename = value
        End Set
    End Property
End Class

File: src/MixNetFx/ADSqlSupportComponents/AdSqlDirectoryPickerDialog.cs
using System.Windows.Forms;

namespace ActiveDev.Data.SqlClient
{
    public partial class ADSqlDirectoryPickerDialog : Form
    {
        private string _sqlPathFilename;

        public DialogResult ShowDialog(string serverConnection, string extensionFilter)
        {
            DirectoryPicker.BeginUpdate();
            DirectoryPicker.ConnectionString = serverConnection;
            DirectoryPicker.ExtensionFilter = extensionFilter;
            DirectoryPicker.EndUpdate();
            return this.ShowDialog();
        }

        public DialogResult ShowDialog(string serverConnection)
        {
            DirectoryPicker.BeginUpdate();
            DirectoryPicker.ConnectionString = serverConnection;
            DirectoryPicker.ExtensionFilter = null;
            DirectoryPicker.EndUpdate();
            return this.ShowDialog();
        }

        private void DirectoryPicker_SelectedFileNodeChanged(object sender, ADFileTreeViewEventArgs e)
        {
            txtPath.Text = e.Node.FullPath;
        }

        private void btnOK_Click(object sender, System.EventArgs e)
        {
            _sqlPathFilename = txtPath.Text;
        }

        private void btnCancel_Click(object sender, System.EventArgs e)
        {
            _sqlPathFilename = null;
        }

        public string SqlPathFilename
        {
            get => _sqlPathFilename;
            set => _sqlPathFilename = value;
        }
    }
}
